### PR TITLE
feat(tools/packages): update command

### PR DESCRIPTION
**Pull request by InteractiveTimmy**
# NOTE

When this pull request is merged, this will publish the `@webex/cli-tools` and `@webex/package-tools` packages to **NPM** for consumption within down-stream clients.

# Description

The scope of the changes in this pull request are to add the `update` command to the `@webex/package-tools` package. This new command allows for consumers of the JS SDK to update their dependencies based on the latest available from `NPMJS`.

# Example

## Before

```jsonc
/* ./package.json */
{
  /* ... */
  "dependencies": {
    /* ... */
    "querystring-es3": "^0.2.1"
  },
  "devDependencies": {
    /* ... */
    "@babel/cli": "^7.17.10",
    "@babel/core": "^7.17.10"
  }
}
```

## Command

```bash
# Input
# package.json script `webex-package-tools` calls `webex-package-tools`
yarn webex-package-tools update --packages @babel/cli querystring-es3 --tag next

# Output
querystring-es3: ^0.2.1 => 1.0.0-0
@babel/cli: ^7.17.10 => 8.0.0-alpha.6
```

## After

```jsonc
/* ./package.json */
{
  /* ... */
  "dependencies": {
    /* ... */
    "querystring-es3": "1.0.0-0" /* updates to eplicit exact */
  },
  "devDependencies": {
    /* ... */
    "@babel/cli": "8.0.0-alpha.6", /* updates all dependency groups */
    "@babel/core": "^7.17.10" /* does not impact non-provided packages */
  }
}
```

# Links

* [Task](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-497350)

### CODE REVIEW

File: packages/tools/cli/package.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
   "name": "@webex/cli-tools",
-  "private": true,
```
Comment (sreenara) [inline packages/tools/cli/package.json:3]: Is there a reason why we are making this also public?

**Review by sreenara — COMMENTED** [review]
@InteractiveTimmy could you please add some examples of the outputs of the update command to the PR description?

File: packages/tools/cli/package.json (hunk 1 above)
Comment (sreenara) [reply packages/tools/cli/package.json:3]: Nevermind, I see `CommandsCommand` being used.

**Comment by sreenara** [comment]
Another general comment @InteractiveTimmy , if the pipeline automatically publishes the current cli-tools and package-tools, it'll do a sync, make them versions 0.0.0-next.0 and publish those versions to NPM. Perhaps we should avoid that and manually publish with version 1.0.0 or something similar?

**Comment by InteractiveTimmy** [comment]
> @InteractiveTimmy could you please add some examples of the outputs of the update command to the PR description?

I have updated the description in this pull request.

> Another general comment @InteractiveTimmy , if the pipeline automatically publishes the current cli-tools and package-tools, it'll do a sync, make them versions 0.0.0-next.0 and publish those versions to NPM. Perhaps we should avoid that and manually publish with version 1.0.0 or something similar?

I do not believe this should be published at version 1.0.0, as this is still an EFT of these packages for consumers. Whenever your team gets a request from a customer that would like this feature, or if you plan on providing this featureset in the core Webex JS SDK utils for external developers, we can then bump to 1.0.0.

**Review by sreenara — APPROVED** [review]
LGTM!

**Comment by sreenara** [comment]
@InteractiveTimmy one note is that this only updates the root level package.json and doesn't recursively run across all packages within the mono repo.

**Comment by InteractiveTimmy** [comment]
> @InteractiveTimmy one note is that this only updates the root level package.json and doesn't recursively run across all packages within the mono repo.

This will update any package that runs the command as a sub-module. So it can be ran from the context of any package or sub-package in a monorepo, if installed at that context.

**Pull request merged**